service/src/test/java/bio/terra/cbas/common/TestMethodUtil.java

```diff
@@ -0,0 +1,122 @@
+package bio.terra.cbas.common;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import bio.terra.cbas.controllers.MethodsApiController;
+import bio.terra.cbas.dao.MethodDao;
+import bio.terra.cbas.dao.MethodVersionDao;
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.dependencies.dockstore.DockstoreService;
+import bio.terra.cbas.dependencies.wes.CromwellService;
+import bio.terra.cbas.model.PostMethodRequest;
+import bio.terra.dockstore.client.ApiException;
+import bio.terra.dockstore.model.ToolDescriptor;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+
+@WebMvcTest
```

Comment:
These tests look more like straight unit tests than MVC (API-level) tests?

Reply:
Without the `@WebMvcTest` annotation the tests were failing as it was receiving the `dockstoreService` mock bean as null. But I am not very familiar with Spring magic and maybe there is something else that can be done to fix the failures instead of adding this particular annotation?

Reply:
Since you're not using this for MVC testing, you don't need to worry about indirectly injecting mocks into the spring context. You can just provide them directly when you call the methods. So, I think you can remove this annotation _and_ the `@MockBean`s below, and when you need a `dockstoreService` object for one of your tests, just create one and configure it as you need it

Reply:
Ah I see. Okay. So I have mocked out the `DockstoreService` class and removed all the unnecessary annotations 👍